src/components/DeadlineCountdown.tsx

```diff
@@ -0,0 +1,224 @@
+import {
+  Text,
+  Flex,
+  Progress,
+  HStack,
+  Button,
+  useColorModeValue,
+} from "@chakra-ui/react";
+import { FaClock } from "react-icons/fa";
+import { differenceInDays, differenceInHours } from "date-fns";
+
+const status_map = [
+  {
+    name: "即將開始",
+    emoji: "⏰",
+    color: "blue.200",
+  },
+  {
+    name: "已開始",
+    emoji: "🏎️",
+    color: "green.200",
+  },
+  {
+    name: "快結束啦!",
+    emoji: "🏃",
+    color: "yellow.200",
+  },
+  {
+    name: "剩不到一天了",
+    emoji: "🥵",
+    color: "red.200",
+  },
+];
+
+const ntu_course_select_url = [
+  "https://if192.aca.ntu.edu.tw/index.php",
+  "https://if177.aca.ntu.edu.tw/index.php",
+];
+
+interface CourseSelectionSchedule {
+  start: number;
+  end: number;
+  name: string;
+  label: string;
+}
+
+const course_select_schedule: CourseSelectionSchedule[] = [
+  {
+    name: "初選第一階段",
```

Comment:
We have to manually change the timestamp every semester right now. But since the whole UI will be redesigned and this component probably will be deleted, I think we don't have to worry about this.